### ISSUE TITLE
Add opus and mka to default audio extensions

### ISSUE THREAD

**Issue by izderadicka** [airsonic/airsonic#603]
## Problem description

Add opus (opus codec in ogg container mime audio/x-opus+ogg) and mka (audio in matroska container mime audio/x-matroska) to default audio extensions, later addition can be problematic.
Opus is becoming more and more popular and it's my favorite audio codec for audiobooks.
Recommendation from Ximph is to use ogg container for opus and above mentioned mime type. 
On Android opus support is limited to matroska container -  thus mka extension is usefull. 
Both extensions are supported on linux desktop for a while.  Opus plays without problem in airsonic (now supported by majority of browsers, FF and Chrome for sure). 

### Steps to reproduce
I bit struggled to get opus supported:
1) I added media dirs
2) Scan media dirs
3) Cannot see opus files in web player  I realize that I need to add them in Settings / Global/ Audio files
4) Added there , rescan media - nothing still not visible
5) Removed media directories , clear db (from Settings page), re-added media directories and rescanned - still nothing -  I guess files stayed in DB as ignored

Finally I had to delete all database (basically by deleting airsonic data forlder), then startt airsonic again and then first add audio files extension in Settings before adding and scanning media folders for first time

## System information

 * **Airsonic version**: *10.0.1-RELEASE – August 23, 2017*
 * **Operating system**: *Linux, Ubuntu 12.04*
 * **Proxy server**: *None*
 * **Client**: *FF 56 or Chromium 61 on Linux Ubuntu 16.04*

**Comment by jooola** [comment]
I think this may need some bigger change than simply adding them by default ( if only a fresh start solves this something might be wrong )

But i guess if we add those 2 extension to this line we should be good to go !
https://github.com/airsonic/airsonic/blob/master/airsonic-main/src/main/java/org/airsonic/player/service/SettingsService.java#L136

@muff1nman little comment on this ? could be a quick fix for next release 😄

**Comment by jooola** [comment]
see #548 which might be related

**Comment by muff1nman** [comment]
Adding them is the easy part. Finding the bug to have this pick up files on a new scan is the harder part ;) We'll keep this open until then.

**Comment by KingDuckZ** [comment]
I have the same problem on 10.4.0, I don't really want to delete my data folder to try the workaround tho.

**Comment by yohan-b** [comment]
This bug is actually pretty simple. It is caused by this optimisation:
```
diff --git a/airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java b/airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java
index a67b08dd..46a4248c 100644
--- a/airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java
+++ b/airsonic-main/src/main/java/org/airsonic/player/service/MediaFileService.java
@@ -383,9 +383,10 @@ public class MediaFileService {
     private void updateChildren(MediaFile parent) {
 
         // Check timestamps.
-        if (parent.getChildrenLastUpdated().getTime() >= parent.getChanged().getTime()) {
-            return;
-        }
+        //if (parent.getChildrenLastUpdated().getTime() >= parent.getChanged().getTime()) {
+       //    LOG.trace("discarded because of timestamps " + parent.getPath());
+        //    return;
+        //}
 
         List<MediaFile> storedChildren = mediaFileDao.getChildrenOf(parent.getPath());
         Map<String, MediaFile> storedChildrenMap = new HashMap<String, MediaFile>();
```
As you can see above, I commented it and the opus files were detected. Another workaround would be to "touch" the parent directories of all opus files. 

The correct way of fixing this would probably be to rename "Scan media folders now" to "Scan modified media folders now (quick scan)" and add "Scan all media folders now (full scan)" which will use an argument "force" to ignore the optimization in updateChildren. When changing the list of extensions, a message would warn the user that he needs to launch a full scan.
Either that or zeroing all parent.getChildrenLastUpdated values when the list of extensions is modified, but that might lead to performance issues for very large libraries.

**Comment by muff1nman** [comment]
Partially solved in in #359
